### ISSUE TITLE
Couchdb:Error while fetching server API version

### ISSUE THREAD

**Issue by axiqia** [apache/incubator-openwhisk#4224]
## Environment details:

* local deployment,native ubuntu16.04
```
 $ docker -v
Docker version 18.09.1, build 4c52b90
```

```
$ pip list
Package                            Version   
---------------------------------- ----------
alabaster                          0.7.11    
anaconda-client                    1.7.2     
anaconda-navigator                 1.9.2     
anaconda-project                   0.8.2     
ansible                            2.5.2     
appdirs                            1.4.3     
argcomplete                        1.9.4     
asn1crypto                         0.24.0    
astroid                            2.0.4     
astropy                            3.0.4     
atomicwrites                       1.2.1     
attrs                              18.2.0    
Automat                            0.7.0     
Babel                              2.6.0     
backcall                           0.1.0     
backports.shutil-get-terminal-size 1.0.0     
bcrypt                             3.1.6     
beautifulsoup4                     4.6.3     
bitarray                           0.8.3     
bkcharts                           0.2       
blaze                              0.11.3    
bleach                             2.1.4     
bokeh                              0.13.0    
boto                               2.49.0    
Bottleneck                         1.2.1     
certifi                            2018.11.29
cffi                               1.11.5    
chardet                            3.0.4     
click                              6.7       
cloudpickle                        0.5.5     
clyent                             1.2.2     
colorama                           0.3.9     
conda                              4.5.11    
conda-build                        3.15.1    
constantly                         15.1.0    
contextlib2                        0.5.5     
CouchDB                            1.2       
cryptography                       2.3.1     
cycler                             0.10.0    
Cython                             0.28.5    
cytoolz                            0.9.0.1   
dask                               0.19.1    
datashape                          0.5.4     
decorator                          4.3.0     
defusedxml                         0.5.0     
distributed                        1.23.1    
docker                             2.2.1     
docker-pycreds                     0.4.0     
docutils                           0.14      
entrypoints                        0.2.3     
et-xmlfile                         1.0.1     
fastcache                          1.0.2     
filelock                           3.0.8     
Flask                              1.0.2     
Flask-Cors                         3.0.6     
gevent                             1.3.6     
glob2                              0.6       
gmpy2                              2.0.8     
greenlet                           0.4.15    
h5py                               2.8.0     
heapdict                           1.0.0     
html5lib                           1.0.1     
httplib2                           0.9.2     
hyperlink                          18.0.0    
idna                               2.8       
imageio                            2.4.1     
imagesize                          1.1.0     
incremental                        17.5.0    
ipykernel                          4.9.0     
ipython                            6.5.0     
ipython-genutils                   0.2.0     
ipywidgets                         7.4.1     
isort                              4.3.4     
itsdangerous                       0.24      
jdcal                              1.4       
jedi                               0.12.1    
jeepney                            0.3.1     
Jinja2                             2.9.6     
jsonschema                         2.6.0     
jupyter                            1.0.0     
jupyter-client                     5.2.3     
jupyter-console                    5.2.0     
jupyter-core                       4.4.0     
jupyterlab                         0.34.9    
jupyterlab-launcher                0.13.1    
keyring                            13.2.1    
kiwisolver                         1.0.1     
lazy-object-proxy                  1.3.1     
llvmlite                           0.24.0    
locket                             0.2.0     
lxml                               4.2.5     
MarkupSafe                         1.0       
matplotlib                         2.2.3     
mccabe                             0.6.1     
mistune                            0.8.3     
mkl-fft                            1.0.4     
mkl-random                         1.0.1     
more-itertools                     4.3.0     
mpmath                             1.0.0     
msgpack                            0.5.6     
multipledispatch                   0.6.0     
navigator-updater                  0.2.1     
nbconvert                          5.4.0     
nbformat                           4.4.0     
networkx                           2.1       
nltk                               3.3       
nose                               1.3.7     
notebook                           5.6.0     
numba                              0.39.0    
numexpr                            2.6.8     
numpy                              1.15.1    
numpydoc                           0.8.0     
odo                                0.5.1     
olefile                            0.46      
openpyxl                           2.5.6     
packaging                          17.1      
pandas                             0.23.4    
pandocfilters                      1.4.2     
paramiko                           2.4.2     
parso                              0.3.1     
partd                              0.3.8     
path.py                            11.1.0    
pathlib2                           2.3.2     
patsy                              0.5.0     
pep8                               1.7.1     
pexpect                            4.6.0     
pickleshare                        0.7.4     
Pillow                             5.2.0     
pip                                18.1      
pkginfo                            1.4.2     
pluggy                             0.7.1     
ply                                3.11      
prometheus-client                  0.3.1     
prompt-toolkit                     1.0.15    
psutil                             5.4.7     
ptyprocess                         0.6.0     
py                                 1.6.0     
pyasn1                             0.4.4     
pyasn1-modules                     0.2.2     
pycodestyle                        2.4.0     
pycosat                            0.6.3     
pycparser                          2.18      
pycrypto                           2.6.1     
pycurl                             7.43.0.2  
pyflakes                           2.0.0     
Pygments                           2.2.0     
pylint                             2.1.1     
PyNaCl                             1.3.0     
pyodbc                             4.0.24    
pyOpenSSL                          18.0.0    
pyparsing                          2.2.0     
PySocks                            1.6.8     
pytest                             3.8.0     
pytest-arraydiff                   0.2       
pytest-astropy                     0.4.0     
pytest-doctestplus                 0.1.3     
pytest-openfiles                   0.3.0     
pytest-remotedata                  0.3.0     
python-dateutil                    2.7.3     
pytz                               2018.5    
PyWavelets                         1.0.0     
PyYAML                             3.13      
pyzmq                              17.1.2    
QtAwesome                          0.4.4     
qtconsole                          4.4.1     
QtPy                               1.5.0     
requests                           2.21.0    
rope                               0.11.0    
ruamel-yaml                        0.15.46   
scikit-image                       0.14.0    
scikit-learn                       0.19.2    
scipy                              1.1.0     
seaborn                            0.9.0     
SecretStorage                      3.1.0     
Send2Trash                         1.5.0     
service-identity                   17.0.0    
setuptools                         40.6.3    
simplegeneric                      0.8.1     
singledispatch                     3.4.0.3   
six                                1.12.0    
snowballstemmer                    1.2.1     
sortedcollections                  1.0.1     
sortedcontainers                   2.0.5     
Sphinx                             1.7.9     
sphinxcontrib-websupport           1.1.0     
spyder                             3.3.1     
spyder-kernels                     0.2.6     
SQLAlchemy                         1.2.11    
statsmodels                        0.9.0     
sympy                              1.2       
tables                             3.4.4     
tblib                              1.3.2     
terminado                          0.8.1     
testpath                           0.3.1     
toolz                              0.9.0     
tornado                            5.1       
tqdm                               4.26.0    
traitlets                          4.3.2     
Twisted                            18.7.0    
unicodecsv                         0.14.1    
urllib3                            1.24.1    
wcwidth                            0.1.7     
webencodings                       0.5.1     
websocket-client                   0.54.0    
Werkzeug                           0.14.1    
wheel                              0.31.1    
widgetsnbextension                 3.4.1     
wrapt                              1.10.11   
xlrd                               1.1.0     
XlsxWriter                         1.1.0     
xlwt                               1.3.0     
zict                               0.1.3     
zope.interface                     4.5.0  
```

```
$ cat db_local.ini    
[db_creds]
db_provider=CouchDB
db_username=whisk_admin
db_password=some_passw0rd
db_protocol=http
db_host=172.17.0.1
db_port=5984

[controller]
db_username=whisk_local_controller0
db_password=some_controller_passw0rd

[invoker]
db_username=whisk_local_invoker0
db_password=some_invoker_passw0rd

```
## Steps to reproduce the issue:
```
ansible-playbook -i environments/local setup.yml  
ansible-playbook -i environments/local couchdb.yml 
```

## Provide the actual results and outputs:

```
$ ansible-playbook -i environments/local couchdb.yml

PLAY [db] *************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ************************************************************************************************************************************************************************************************************************
Saturday 19 January 2019  16:09:41 +0800 (0:00:00.165)       0:00:00.165 ****** 
ok: [172.17.0.1]

TASK [couchdb : set the coordinator to the first node] ****************************************************************************************************************************************************************************************
Saturday 19 January 2019  16:09:43 +0800 (0:00:02.056)       0:00:02.222 ****** 
ok: [172.17.0.1]

TASK [couchdb : Set the volumes] **************************************************************************************************************************************************************************************************************
Saturday 19 January 2019  16:09:43 +0800 (0:00:00.090)       0:00:02.312 ****** 
ok: [172.17.0.1]

TASK [couchdb : check if db credentials are valid for CouchDB] ********************************************************************************************************************************************************************************
Saturday 19 January 2019  16:09:43 +0800 (0:00:00.089)       0:00:02.402 ****** 
skipping: [172.17.0.1]

TASK [couchdb : check for persistent disk] ****************************************************************************************************************************************************************************************************
Saturday 19 January 2019  16:09:43 +0800 (0:00:00.131)       0:00:02.534 ****** 
skipping: [172.17.0.1]

TASK [couchdb : set the volume_dir] ***********************************************************************************************************************************************************************************************************
Saturday 19 January 2019  16:09:44 +0800 (0:00:00.059)       0:00:02.593 ****** 
skipping: [172.17.0.1]

TASK [couchdb : include_tasks] ****************************************************************************************************************************************************************************************************************
Saturday 19 January 2019  16:09:44 +0800 (0:00:00.059)       0:00:02.653 ****** 
skipping: [172.17.0.1]

TASK [couchdb : set the erlang cookie volume] *************************************************************************************************************************************************************************************************
Saturday 19 January 2019  16:09:44 +0800 (0:00:00.118)       0:00:02.772 ****** 
skipping: [172.17.0.1]

TASK [couchdb : (re)start CouchDB from 'apache/couchdb:2.1 '] *********************************************************************************************************************************************************************************
Saturday 19 January 2019  16:09:44 +0800 (0:00:00.124)       0:00:02.896 ****** 
fatal: [172.17.0.1]: FAILED! => {"changed": false, "msg": "Error connecting: Error while fetching server API version: Timeout value connect was Timeout(connect=60, read=60, total=None), but it must be an int or float."}

Error connecting: Error while fetching server API version: Timeout value
connect was Timeout(connect=60, read=60, total=None), but it must be an int or
float.

PLAY RECAP ************************************************************************************************************************************************************************************************************************************
172.17.0.1                 : ok=3    changed=0    unreachable=0    failed=1   

Saturday 19 January 2019  16:09:45 +0800 (0:00:01.665)       0:00:04.562 ****** 
=============================================================================== 
Gathering Facts ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 2.06s
couchdb : (re)start CouchDB from 'apache/couchdb:2.1 '  -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 1.67s
couchdb : check if db credentials are valid for CouchDB -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.13s
couchdb : set the erlang cookie volume ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.12s
couchdb : include_tasks ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.12s
couchdb : set the coordinator to the first node ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.09s
couchdb : Set the volumes -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.09s
couchdb : set the volume_dir ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.06s
couchdb : check for persistent disk ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.06
```

**Comment by axiqia** [comment]
```
$ docker run apache/couchdb:2.1
****************************************************
WARNING: CouchDB is running in Admin Party mode.
         This will allow anyone with access to the
         CouchDB port to access your database. In
         Docker's default configuration, this is
         effectively any other container on the same
         system.
         Use "-e COUCHDB_USER=admin -e COUCHDB_PASSWORD=password"
         to set it in "docker run".
****************************************************
[info] 2019-01-19T12:24:11.916806Z nonode@nohost <0.7.0> -------- Application couch_log started on node nonode@nohost
[info] 2019-01-19T12:24:11.921465Z nonode@nohost <0.7.0> -------- Application folsom started on node nonode@nohost
[info] 2019-01-19T12:24:11.959820Z nonode@nohost <0.7.0> -------- Application couch_stats started on node nonode@nohost
[info] 2019-01-19T12:24:11.960002Z nonode@nohost <0.7.0> -------- Application khash started on node nonode@nohost
[info] 2019-01-19T12:24:11.967989Z nonode@nohost <0.7.0> -------- Application couch_event started on node nonode@nohost
[info] 2019-01-19T12:24:11.973535Z nonode@nohost <0.7.0> -------- Application ibrowse started on node nonode@nohost
[info] 2019-01-19T12:24:11.978533Z nonode@nohost <0.7.0> -------- Application ioq started on node nonode@nohost
[info] 2019-01-19T12:24:11.978714Z nonode@nohost <0.7.0> -------- Application mochiweb started on node nonode@nohost
[info] 2019-01-19T12:24:11.986264Z nonode@nohost <0.195.0> -------- Apache CouchDB 2.1.2 is starting.

[info] 2019-01-19T12:24:11.986428Z nonode@nohost <0.196.0> -------- Starting couch_sup
[notice] 2019-01-19T12:24:12.019276Z nonode@nohost <0.82.0> -------- config: [couchdb] uuid set to 3e46cfcdf857767cca67ece2441807dd for reason nil
[info] 2019-01-19T12:24:12.890763Z nonode@nohost <0.201.0> -------- open_result error {not_found,no_db_file} for _users
[info] 2019-01-19T12:24:13.086760Z nonode@nohost <0.195.0> -------- Apache CouchDB has started. Time to relax.

[info] 2019-01-19T12:24:13.087063Z nonode@nohost <0.195.0> -------- Apache CouchDB has started on http://any:5986/
[info] 2019-01-19T12:24:13.087330Z nonode@nohost <0.7.0> -------- Application couch started on node nonode@nohost
[info] 2019-01-19T12:24:13.087511Z nonode@nohost <0.7.0> -------- Application ets_lru started on node nonode@nohost
[info] 2019-01-19T12:24:13.121468Z nonode@nohost <0.260.0> -------- Opening index for db: _users idx: _design/_auth sig: "3e823c2a4383ac0c18d4e574135a5b08"
[info] 2019-01-19T12:24:13.128920Z nonode@nohost <0.7.0> -------- Application rexi started on node nonode@nohost
[info] 2019-01-19T12:24:13.140475Z nonode@nohost <0.201.0> -------- open_result error {not_found,no_db_file} for _nodes
[warning] 2019-01-19T12:24:13.140525Z nonode@nohost <0.277.0> -------- creating missing database: _nodes
[info] 2019-01-19T12:24:13.281645Z nonode@nohost <0.201.0> -------- open_result error {not_found,no_db_file} for _dbs
[warning] 2019-01-19T12:24:13.281714Z nonode@nohost <0.290.0> -------- creating missing database: _dbs
[warning] 2019-01-19T12:24:13.281791Z nonode@nohost <0.289.0> -------- creating missing database: _dbs
[info] 2019-01-19T12:24:13.353158Z nonode@nohost <0.7.0> -------- Application mem3 started on node nonode@nohost
[info] 2019-01-19T12:24:13.353294Z nonode@nohost <0.7.0> -------- Application fabric started on node nonode@nohost
[info] 2019-01-19T12:24:13.387525Z nonode@nohost <0.7.0> -------- Application chttpd started on node nonode@nohost
[error] 2019-01-19T12:24:13.400727Z nonode@nohost emulator -------- Error in process <0.327.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:24:13.400778Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[info] 2019-01-19T12:24:13.405523Z nonode@nohost <0.7.0> -------- Application couch_index started on node nonode@nohost
[info] 2019-01-19T12:24:13.405703Z nonode@nohost <0.7.0> -------- Application couch_mrview started on node nonode@nohost
[info] 2019-01-19T12:24:13.405784Z nonode@nohost <0.7.0> -------- Application couch_plugins started on node nonode@nohost
[notice] 2019-01-19T12:24:13.438933Z nonode@nohost <0.82.0> -------- config: [features] scheduler set to true for reason nil
[info] 2019-01-19T12:24:13.458547Z nonode@nohost <0.201.0> -------- open_result error {not_found,no_db_file} for _replicator
[notice] 2019-01-19T12:24:13.522375Z nonode@nohost <0.343.0> -------- creating replicator ddoc <<"_replicator">>
[info] 2019-01-19T12:24:13.586146Z nonode@nohost <0.7.0> -------- Application couch_replicator started on node nonode@nohost
[info] 2019-01-19T12:24:13.600506Z nonode@nohost <0.7.0> -------- Application couch_peruser started on node nonode@nohost
[info] 2019-01-19T12:24:13.615321Z nonode@nohost <0.7.0> -------- Application ddoc_cache started on node nonode@nohost
[info] 2019-01-19T12:24:13.640846Z nonode@nohost <0.7.0> -------- Application global_changes started on node nonode@nohost
[info] 2019-01-19T12:24:13.640899Z nonode@nohost <0.7.0> -------- Application jiffy started on node nonode@nohost
[info] 2019-01-19T12:24:13.652015Z nonode@nohost <0.7.0> -------- Application mango started on node nonode@nohost
[info] 2019-01-19T12:24:13.664674Z nonode@nohost <0.7.0> -------- Application setup started on node nonode@nohost
[info] 2019-01-19T12:24:13.664741Z nonode@nohost <0.7.0> -------- Application snappy started on node nonode@nohost
[error] 2019-01-19T12:24:18.401486Z nonode@nohost emulator -------- Error in process <0.464.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:24:18.401543Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[notice] 2019-01-19T12:24:18.418133Z nonode@nohost <0.347.0> -------- couch_replicator_clustering : cluster stable
[notice] 2019-01-19T12:24:18.425247Z nonode@nohost <0.363.0> -------- Started replicator db changes listener <0.465.0>
[error] 2019-01-19T12:24:23.402453Z nonode@nohost emulator -------- Error in process <0.564.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:24:23.402524Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[notice] 2019-01-19T12:24:28.129103Z nonode@nohost <0.267.0> -------- rexi_server : cluster stable
[notice] 2019-01-19T12:24:28.129098Z nonode@nohost <0.270.0> -------- rexi_buffer : cluster stable
[error] 2019-01-19T12:24:28.403411Z nonode@nohost emulator -------- Error in process <0.631.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:24:28.403454Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:24:33.404434Z nonode@nohost emulator -------- Error in process <0.728.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:24:33.404529Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:24:38.405460Z nonode@nohost emulator -------- Error in process <0.793.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:24:38.405550Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:24:43.406448Z nonode@nohost emulator -------- Error in process <0.890.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:24:43.406537Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:24:48.407465Z nonode@nohost emulator -------- Error in process <0.955.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:24:48.407538Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:24:53.408443Z nonode@nohost emulator -------- Error in process <0.1052.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:24:53.408519Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:24:58.409464Z nonode@nohost emulator -------- Error in process <0.1117.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:24:58.409532Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:25:03.410444Z nonode@nohost emulator -------- Error in process <0.1214.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:25:03.410520Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:25:08.411470Z nonode@nohost emulator -------- Error in process <0.1279.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:25:08.411550Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:25:13.412461Z nonode@nohost emulator -------- Error in process <0.1376.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:25:13.412542Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:25:18.413442Z nonode@nohost emulator -------- Error in process <0.1441.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:25:18.413496Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:25:23.414392Z nonode@nohost emulator -------- Error in process <0.1538.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:25:23.414465Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:25:28.415481Z nonode@nohost emulator -------- Error in process <0.1603.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 
[notice] 2019-01-19T12:25:28.415557Z nonode@nohost <0.326.0> -------- chttpd_auth_cache changes listener died database_does_not_exist at mem3_shards:load_shards_from_db/6(line:403) <= mem3_shards:load_shards_from_disk/1(line:378) <= mem3_shards:load_shards_from_disk/2(line:407) <= mem3_shards:for_docid/3(line:91) <= fabric_doc_open:go/3(line:38) <= chttpd_auth_cache:ensure_auth_ddoc_exists/2(line:187) <= chttpd_auth_cache:listen_for_changes/1(line:134)
[error] 2019-01-19T12:25:33.416408Z nonode@nohost emulator -------- Error in process <0.1700.0> with exit value: {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,378}]},{mem3_shards,load_shards_from_disk... 

```

@csantanapr ,is it right ？

**Comment by csantanapr** [comment]
That’s right 
It looks like the error “Error while fetching server API version” is not couchdb. 
I think the error is from the docker client python module use by ansible to start the docker container.

**Comment by csantanapr** [comment]
I think is related to ansible and the host where docker is running. 
This issue might help you on what’s going with your setup localhost https://github.com/ansible/ansible-modules-core/issues/5685

**Comment by axiqia** [comment]
According to  this [issue]( https://github.com/requests/requests/issues/4160) , I try to switch `requests` to version `2.18.1`, but it didn't help.

**Comment by csantanapr** [comment]
Well there is something wrong about a network timeout on that ansible step using the docker plugin to start the container any container. 
Not a specific problem about OpenWhisk setup. 

It might be that DOCKER_HOST needs to be set. 

Are you using docker machine or normal docker engine on host?

**Comment by axiqia** [comment]
I use  normal docker engine and the verison is 
```
 $ docker -v
Docker version 18.09.1, build 4c52b90
```

**Comment by axiqia** [comment]
Maybe I miss something, I tried to install without `sudo`  so I had deleted `sudo` in openwhisk/tools/ubuntu-setup/ansible.sh.

Then I got a fatal:
```
ansible-playbook -i environments/local couchdb.yml 
....
fatal: [172.17.0.1]: FAILED! => {"changed": false, "msg": "Failed to import docker-py - No module named requests.exceptions. Try `pip install docker-py`"}
```

Reference to this [question](https://github.com/ansible/ansible-modules-core/issues/4246#issuecomment-268361256), I changed` ansible_python_interpreter` and I got the problem above.

I can't resolve it,  so I add `sudo` to ansible.sh file.
 Although I don't know why, but it works.
